opentasks-provider/src/main/java/org/dmfs/provider/tasks/TaskDatabaseHelper.java

```diff
@@ -166,6 +168,35 @@
             + " JOIN " + Tables.LISTS + " ON (" + Tables.TASKS + "." + TaskContract.Tasks.LIST_ID + "=" + Tables.LISTS + "." + TaskContract.Tasks._ID + ")"
             + " JOIN " + Tables.INSTANCES + " ON (" + Tables.TASKS + "." + TaskContract.Tasks._ID + "=" + Tables.INSTANCES + "." + TaskContract.Instances.TASK_ID + ");";
 
+    /**
+     * SQL command to create a view that combines task instances with some data from the list they belong to. This replaces the task DTSTART and DUE
+     * values with respective values of the instance.
+     * <p>
+     * This is the instances view as seen by the content provider clients.
+     */
+    private final static String SQL_CREATE_INSTANCE_CLIENT_VIEW = "CREATE VIEW " + Tables.INSTANCE_CLIENT_VIEW + " AS SELECT "
+            + Tables.INSTANCES + ".*, "
+            // override task due and start with the instance values
+            + Tables.INSTANCES + "." + TaskContract.Instances.INSTANCE_START + " as " + Tasks.DTSTART + ", "
+            + Tables.INSTANCES + "." + TaskContract.Instances.INSTANCE_DUE + " as " + Tasks.DUE + ", "
+            // override task duration with null, we already have a due
+            + "null as " + Tasks.DURATION + ", "
+            // override recurrence values with null, instances themselves are not recurring
+            + "null as " + Tasks.RRULE + ", "
+            + "null as " + Tasks.RDATE + ", "
+            + "null as " + Tasks.EXDATE + ", "
+            + Tables.TASKS + ".*, "
+            + Tables.LISTS + "." + Tasks.ACCOUNT_NAME + ", "
+            + Tables.LISTS + "." + Tasks.ACCOUNT_TYPE + ", "
+            + Tables.LISTS + "." + Tasks.LIST_OWNER + ", "
+            + Tables.LISTS + "." + Tasks.LIST_NAME + ", "
+            + Tables.LISTS + "." + Tasks.LIST_ACCESS_LEVEL + ", "
+            + Tables.LISTS + "." + Tasks.LIST_COLOR + ", "
+            + Tables.LISTS + "." + Tasks.VISIBLE
+            + " FROM " + Tables.TASKS
+            + " JOIN " + Tables.LISTS + " ON (" + Tables.TASKS + "." + TaskContract.Tasks.LIST_ID + "=" + Tables.LISTS + "." + TaskContract.Tasks._ID + ")"
```

Comment:
At the end of the first join, shouldn't `TaskContract.Tasks._ID` be `TaskContract.TaskLists._ID` instead? I know it's effectively the same, but just for clarity.

Reply:
Right, I'll fix that.